core/src/main/java/org/infinispan/distribution/ch/impl/RESPHashFunctionPartitioner.java

```diff
@@ -0,0 +1,46 @@
+package org.infinispan.distribution.ch.impl;
+
+import org.infinispan.commons.hash.CRC16;
+import org.infinispan.commons.hash.Hash;
+import org.infinispan.commons.util.Util;
+
+/**
+ * Implementation of {@link HashFunctionPartitioner} using {@link CRC16} for use with RESP.
+ * Note that RESP only has 14 bits worth of slots and uses remainder based bucket allocation.
+ * Due to this we have to treat it as having only 14 bits for a hash
+ *
+ * @since 15.0
+ * @see HashFunctionPartitioner
+ */
+public class RESPHashFunctionPartitioner extends HashFunctionPartitioner {
```

Comment:
I thought about moving to RESP, but since it is available for configuration would likely break embedded, requiring a dependency on the RESP module to work.

Reply:
Yeah, I don't think I was able to figure this out originally. I had planned this to be CRC16 only so it wasn't tied to RESP. But unfortunately the `bitsToUse` method messes that up as CRC16 uses all 16 and RESP only uses 14 :(